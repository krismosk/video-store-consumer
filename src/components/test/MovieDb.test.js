import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react';
import MovieDB from '../MovieDb';

const movieDB = <MovieDB 
  id={ 1337 }
  title={'test title'}
  image={'test image'}
  overview={'test overview'}
  releaseDate={'test date'}
  externalId={1}
  addMovieCallback={() => { }}
/>;

describe('MovieDB', () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render( movieDB );
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
  test("it will render the proper data for the movie", () => {
    const container = render(movieDB)
    expect(container.getByText(/test title/)).toBeDefined();
    expect(container.getByText(/test overview/)).toBeDefined();
    expect(container.getByText(/test date/)).toBeDefined();
  });
  test('the findMovie function is called when the select button is clicked on', () => {
    const addMovieCallback = jest.fn();
    const container = render(<MovieDB 
      id={ 1337 }
      title={'test title'}
      image={'test image'}
      overview={'test overview'}
      releaseDate={'test date'}
      inventory={1}
      addMovieCallback={addMovieCallback}
    />);

    const selectButton = container.getByText(/Add to Rental Library/);
    selectButton.click();
    expect(addMovieCallback).toHaveBeenCalled();
  })
});