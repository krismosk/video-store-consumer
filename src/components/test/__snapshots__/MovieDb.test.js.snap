// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDB it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="MovieCard"
  >
    <div>
      <img
        class="searchIcon"
        src="test image"
      />
    </div>
    <div>
      <h3
        class="searchMovieTitle"
      >
        test title
      </h3>
      <p
        class="searchDate"
      >
        test date
      </p>
      <p
        class="searchOverview"
      >
        test overview
      </p>
    </div>
    <div>
      <button
        class="button searchAdd"
        type="button"
      >
        Add to Rental Library
      </button>
    </div>
  </section>
</DocumentFragment>
`;
